nservicebus/upgrades/azure-functions-service-bus-in-process-isolated-worker.md

```diff
@@ -0,0 +1,163 @@
+
+### Modify `HttpSender.cs`
+
+Remove the following references:
+
+- `Microsoft.AspNetCore.Http`
+- `Microsoft.AspNetCore.Mvc`
+- `Microsoft.Azure.WebJobs.Http`
+- `Microsoft.Azure.WebJobs`
+- `Microsoft.Azure.WebJobs.Extensions.Http`
+
+Add the following references
+
+- `Microsoft.Azure.Functions.Worker`
+- `Microsoft.Azure.Functions.Worker.Extensions.Http`
+
+### Update the `Run` method signature and logic
+
+- Change from `Task<IActionResult>` to `Task<HttpResponseData>`
+- Change `ExecutionContext` parameter to `FunctionContext`
+- The `functionEndpoint.send` method now does not take the logger as a parameter.
+
+```csharp
+[Function("HttpSender")]
+public async Task<HttpResponseData> Run(
+    [HttpTrigger(AuthorizationLevel.Function, "get", "post", Route = null)]
+    HttpRequestData req,
+    FunctionContext executionContext)
+{
+    var logger = executionContext.GetLogger<HttpSender>();
+    logger.LogInformation("C# HTTP trigger function received a request.");
+
+    var sendOptions = new SendOptions();
+    sendOptions.RouteToThisEndpoint();
+
+    await functionEndpoint.Send(new TriggerMessage(), sendOptions, executionContext);
+
+    var r = req.CreateResponse(HttpStatusCode.OK);
+    await r.WriteStringAsync($"{nameof(TriggerMessage)} sent.");
+    return r;
+}
+```
+
+## Update host configuration
+
+Utilizing the minimal hosting model available in .Net 6.0, the `startup.cs` file can be merged with the `program.cs` file to register services.
+
+### Simple example
+
+```csharp
+using Microsoft.Extensions.Hosting;
+using NServiceBus;
+using System.Threading.Tasks;
+
+[assembly:NServiceBusTriggerFunction("ASBWorkerEndpoint")]
+
+public class Program
+{
+    public static Task Main()
+    {
+        var host = new HostBuilder()
+            .ConfigureFunctionsWorkerDefaults()
+            .UseNServiceBus()
+            .Build();
+
+        return host.RunAsync();
+    }
+}
+```
+
+### With dependency injection
+
+```csharp
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Configuration;
+using NServiceBus;
+using System.Threading.Tasks;
+
+[assembly: NServiceBusTriggerFunction("ASBWorkerEndpoint")]
+
+public class Program
+{
+    public static Task Main()
+    {
+        var host = new HostBuilder()
+            .ConfigureFunctionsWorkerDefaults(builder =>
+            {
+                // Register custom service in the container
+                builder.Services.AddSingleton(serviceProvider =>
+                {
+                    var configurationRoot = serviceProvider.GetRequiredService<IConfiguration>();
+                    var customComponentInitializationValue = configurationRoot.GetValue<string>("CustomComponentValue");
+
+                    return new CustomComponent(customComponentInitializationValue);
+                });
+            })
+            .UseNServiceBus()
+            .Build();
+
+        return host.RunAsync();
+    }
+}
+```
```

Comment:
```suggestion
```## Update local.settings.json

Update the key value for "FUNCTIONS_WORKER_RUNTIME": "dotnet", to "FUNCTIONS_WORKER_RUNTIME": "dotnet-isolated",
```

Reply:
This is addressed in Microsoft's guide: https://learn.microsoft.com/en-us/azure/azure-functions/migrate-dotnet-to-isolated-model?tabs=net8#localsettingsjson-file